src/pybind/mgr/cephadm/services/ingress.py

```diff
@@ -17,10 +17,18 @@ class IngressService(CephService):
     TYPE = 'ingress'
     MAX_KEEPALIVED_PASS_LEN = 8
 
-    def primary_daemon_type(self) -> str:
+    def primary_daemon_type(self, spec: Optional[ServiceSpec] = None) -> str:
+        if spec:
+            ispec = cast(IngressSpec, spec)
+            if hasattr(spec, 'keepalive_only') and ispec.keepalive_only:
+                return 'keepalived'
         return 'haproxy'
 
-    def per_host_daemon_type(self) -> Optional[str]:
+    def per_host_daemon_type(self, spec: Optional[ServiceSpec] = None) -> Optional[str]:
+        if spec:
+            ispec = cast(IngressSpec, spec)
+            if hasattr(spec, 'keepalive_only') and ispec.keepalive_only:
+                return None
```

Comment:
I think It'd be good to add some comment explaining the reason behind this behaviour as in both case we are checking for `keepalalived` but in some cases we return None.

Reply:
added comments